corehq/apps/data_interfaces/static/data_interfaces/js/case_dedupe_main.js

```diff
@@ -14,19 +14,156 @@ hqDefine("data_interfaces/js/case_dedupe_main", [
     casePropertyInput,
     CaseRuleCriteria
 ) {
+
+    /*
+    PropertyManager exists to make sure that available properties are updated and unique.
+    For example, if a case type contains A, B, and C properties, this class ensures
+    that if C is selected, further options will only show A and B.
+    */
+    const PropertyManager = function (caseType, selectedProperties, initialPropertyMap) {
+        const self = {};
+        self.caseType = caseType;
+        self.selectedProperties = selectedProperties;
```

Comment:
might be good to note somewhere that `self.selectProperties` must be an `observableArray` and `self.caseType` is an `observable`

Reply:
26d8ba4f41466fab3dd362e04fae5723102a2a20